Project/core/src/main/java/Model/NormalCell.java

```diff
@@ -41,4 +43,28 @@ public NormalCell(Feature[] features) {
     public Feature[] getFeatures() {
         return features;
     }
+
+    @Override
+    public String toString() {
+        return "NormalCell{" +
+                "features=" + Arrays.toString(features) +
+                ", players=" + players +
+                ", tokens=" + tokens +
+                '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!super.equals(o))
+            return false;
+        NormalCell cell = (NormalCell) o;
+        return Arrays.deepEquals(features, cell.features);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = super.hashCode();
+        result = 31 * result + Arrays.hashCode(features);
```

Comment:
I "automated" the creation of the three functions toString(), equals() and hashCode() using ChatGPT since it was a "simple" and repetitive task that would have taken super long time to do manually, but it still was tedious work since I had to doublecheck and make small changes. But basically the hashCode was the weirdest part and also least important (since it's not currently used but at the same time is connected to the equals() function), but I guess you can create a hashCode however you want, but it should be as random as possible to create an even distribution of the hashed objects, so I guess this is just to make it a bit more random than just an addition.